packages/fxa-settings/src/pages/Signin/utils.ts

```diff
@@ -213,16 +219,28 @@ export async function handleNavigation(navigationOptions: NavigationOptions) {
     navigationOptions.syncHidePromoAfterLogin = true;
   }
 
+  // When a session is unverified, we need to redirect to the appropriate page depending on status of
+  // the account and the integration being used.
+  // The following cases are handled:
+  // 1. Users that don't have a verified email always get redirected to confirm signup
+  // 2. Users that have a TOTP always get redirected to confirm TOTP
+  // 3. OAuthNative integrations always get redirected to confirm email
+  // 4. Integrations that want two-step authentication always get redirected to confirm email, before
+  //    setting up TOTP
+  // 5. Non OAuthNative (ie Relay, Monitor) are redirected to RP
```

Comment:
Nit, instead of "Non OAuthNative" maybe call it `OAuthWeb`? Thanks for the comment!